flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicTableFactory.java

```diff
@@ -374,7 +380,41 @@ protected KafkaDynamicSource createKafkaTableSource(
                 specificStartupOffsets,
                 startupTimestampMillis,
                 false,
-                tableIdentifier);
+                tableIdentifier,
+                null);
+    }
+
+    protected KafkaDynamicSource createKafkaTableSource(
```

Comment:
I believe it would be better to not clone but modify the current constructor and make `sourceParallelism`nullable